src/Text/Pandoc/Readers/HTML.hs

```diff
@@ -786,18 +786,28 @@ pSvg = do
 pCodeWithClass :: PandocMonad m => Text -> Text -> TagParser m Inlines
 pCodeWithClass name class' = try $ do
   TagOpen open attr' <- pSatisfy $ tagOpen (== name) (const True)
-  result <- manyTill pAny (pCloses open)
   let (ids,cs,kvs) = toAttr attr'
       cs'          = class' : cs
-  return . B.codeWith (ids,cs',kvs) .
-    T.unwords . T.lines . innerText $ result
+  code open (ids,cs',kvs)
 
 pCode :: PandocMonad m => TagParser m Inlines
 pCode = try $ do
   (TagOpen open attr') <- pSatisfy $ tagOpen (`elem` ["code","tt"]) (const True)
   let attr = toAttr attr'
-  result <- manyTill pAny (pCloses open)
-  return $ B.codeWith attr $ T.unwords $ T.lines $ innerText result
+  code open attr
+
+code :: PandocMonad m => Text -> Attr -> TagParser m Inlines
+code open attr = do
+  exts <- getOption readerExtensions
+  -- Don't use `formatCode` in an epub because it can incorrectly parse certain
+  -- elements
```

Comment:
That's weird. Could you include an example to show what could go wrong?

Reply:
Done. There may be a way to fix this issue and still use `formatCode`, but I'm not very familiar with the epub format and the fact that I can't see the test input makes it hard for me to troubleshoot. Apologies if I'm missing something.